goosebit/models.py

```diff
@@ -38,3 +45,65 @@ class Rollout(Model):
     paused = fields.BooleanField(default=False)
     success_count = fields.IntField(default=0)
     failure_count = fields.IntField(default=0)
+
+
+class FirmwareCompatibility(Model):
+    id = fields.IntField(primary_key=True)
+    hw_model = fields.CharField(max_length=255)
+    hw_revision = fields.CharField(max_length=255)
+
+
+class FirmwareUpdate(Model):
```

Comment:
I would just call it `Firmware` as it does not have to be an update, it could technically also be a downgrade (if we would start to allow that) or the initial image.

Reply:
The rollout model needs to be adjusted as well - instead of hw_model, hw_revision, fw_file it only should reference a firmware.

This then also means work in Manager.get_rollout to query the correct firmware.

Reply:
> The rollout model needs to be adjusted as well - instead of hw_model, hw_revision, fw_file it only should reference a firmware.
> 
> This then also means work in Manager.get_rollout to query the correct firmware.

This can get really tricky, if I link `Firmware` directly then all the stuff that wants to check for rollout that uses the device as a backup need to return `Firmware`, which means `fw_file` on `Device` needs to be replaced with `Firmware`, but then that gets rid of `latest` and `pinned`, which can make things pretty complicated.  Let me know what you guys think here.

Reply:
Yes, it triggers a few follow-up changes. I have a branch where I introduced `update_mode` on Device (enum with `latest`, `rollout`, `assigned`, and `pinned`), so that `fw_file` could be changed to `firmware` on Device.

Will re-implement those changes on top of this branch to see if it makes sense.